src/main/webapp/app/entities/iris/iris.model.ts

```diff
@@ -0,0 +1,42 @@
+import dayjs from 'dayjs';
+import { User } from 'app/core/user/user.model';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+
+export class IrisSession {
+    id: number;
+    exercise?: ProgrammingExercise;
+    user?: User;
+    messages: IrisMessage[];
+}
+
+export enum IrisSender {
+    SERVER = 'server',
+    USER = 'user',
+    SYSTEM = 'system',
+}
+
+export class IrisServerMessage {
+    id: number;
+    sender: IrisSender.SERVER;
+    content: IrisMessageContent[];
+    sentAt: dayjs.Dayjs;
+    helpful?: boolean;
+}
+
+export class IrisClientMessage {
+    id?: number;
+    sender: IrisSender.USER;
+    content: IrisMessageContent;
```

Comment:
I think BE doesn't distinguish between client message and server message and will [always a list](https://github.com/ls1intum/Artemis/blob/efe8ecd2f6ecad9528362e49a464444faad74e99/src/main/java/de/tum/in/www1/artemis/domain/iris/IrisMessage.java#L45-L47)
```suggestion
    content: IrisMessageContent[];
```